Assets/Scripts/Extentions/VectorExtensions.cs

```diff
@@ -160,9 +171,12 @@ public static Vector3 SeparateZ(this Vector3 v, Vector3 f)
         /// <param name="v">This vector3.</param>
         /// <param name="f">The f Vector3.</param>
         /// <returns>A new Vector3 that is multiplied.</returns>
-        public static Vector3 MultipleVector(this Vector3 v, Vector3 f)
+        public static ref Vector3 MultipleVector(ref this Vector3 v, Vector3 f)
```

Comment:
```suggestion
        public static ref Vector3 MultiplyVector(ref this Vector3 v, Vector3 f)
```

Reply:
Also I think it kinda defeats the point to add the "Vector" part to the method name. Since this will already be called directly on Vectors anyway.